mathesar_ui/src/utils/errors.ts

```diff
@@ -76,4 +76,16 @@ export class ApiMultiError extends Error {
   }
 }
 
+/**
+ * This is meant to be a build time error
+ * Generally use to make sure of exhaustive conditions handlint
+ */
+export class UnhandledError extends Error {
+  constructor(value: never, message: string) {
```

Comment:
I can see the usefulness of `UnhandledError` so I'm okay with keeping it.

But why does `value` need to be `never` here? More broadly, I don't see the usefulness of the `value` argument all. I might want to throw an `UnhandledError` even if I don't have a value. Or I might have an object as a value. From the calling site, it's not clear what happens with `value` when I pass it into the constructor. Witt the code as-is, if I pass in an object, the error message will display "[object Object]" but that behavior is not clear from the calling site.

I'd be inclined to do:

```ts
export class UnhandledError extends Error {
  constructor(message: string) {
    super(message);
    this.name = "Unhandled Error";
  }
}
```

That design forces the caller to stringify their value (as they see fit, depending on the type of the value) before instantiating the `UnhandledError` object. It also sets the `name` property on the instance so that `String(new UnhandledError("Foo"));` will give `"Unhandled Error: Foo"` instead of `"Error: Unhandled Error: Foo."`

Reply:
@seancolsen `value` being `never` is what makes this error class useful. For example check this case here: 

```typescript
let mode: 'read' | 'edit' = 'read';
  function toggleMode(): undefined {
    switch (mode) {
      case 'read':
        mode = 'edit';
        break;
      case 'edit':
        mode = 'read';
        break;
      default:
        throw new UnhandledError(mode, 'ColumnType');
    }
    return undefined;
  }
```

If tomorrow someones adds a new possible value for `mode` and forgets to handle it inside the `toggleMode` function, the type of `mode` in line `throw new UnhandledError(mode, 'ColumnType');` will not be `never` which is expected hence forcing TS to show build time error.

Reply:
_Ohhhhh_ -- I get it now. 💡 Cool!!

Very clever. I like this!

The name `UnhandledError` threw me off because it's so general. I've opened #1618 to propose some minor clean up which improves the name.